public/pages/workflow_detail/components/header.tsx

```diff
@@ -79,33 +92,23 @@ export function WorkflowDetailHeader(props: WorkflowDetailHeaderProps) {
     }
   }, [props.workflow]);
 
-  const { TopNavMenu, HeaderControl } = getNavigationUI();
-  const { setAppRightControls } = getApplication();
-  const {
-    chrome: { setHeaderVariant },
-  } = getCore();
-  const uiSettings = getUISettings();
-  const showActionsInHeader = uiSettings.get('home:useNewHomePage');
-
+  // When NewHomePage is enabled, use 'application' HeaderVariant; otherwise, use 'page' HeaderVariant (default).
   useEffect(() => {
-    if (showActionsInHeader) {
+    if (SHOW_ACTIONS_IN_HEADER) {
       setHeaderVariant?.(HeaderVariant.APPLICATION);
     }
-
     return () => {
       setHeaderVariant?.();
     };
-  }, [setHeaderVariant, showActionsInHeader]);
+  }, [setHeaderVariant, SHOW_ACTIONS_IN_HEADER]);
```

Comment:
Still bit confused on why it is returning some function on lines 100-102. Shouldn't this just be setting the header variant based on `SHOW_ACTIONS_IN_HEADER`? How about:
```
    if (SHOW_ACTIONS_IN_HEADER) {
      setHeaderVariant?.(HeaderVariant.APPLICATION);
    } else {
      setHeaderVariant?.()
    }
```

Or, do we even need to set it at all in the `else` case?

Reply:
discussed offline, existing logic is fine, as the return fn is just the cleanup fn for when the component unmounts.